### PR TITLE
Add basis for testing of our Spectral API linting

### PR DESCRIPTION
**Pull request by jamietanna**
As we start to produce API linting rules for our cross-government public
APIs, we should write tests to validate that the rules are correct
before we ship them out and potentially break folks' pipelines.

To do this:

- we can utilise the upstream package `spectral-test-harness` to make it
  easier for us to validate our rules
- we need to add a workaround for `Module not found` errors with
  `nimma`
- we add a couple of base tests to validate the minimum requirement for
  our OpenAPI 3 specifications to pass, based on the recommended
  Spectral OpenAPI specification rules

### CODE REVIEW

File: govuk-public-api-rules/test/rules.test.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const { retrieveDocument, setupSpectral, getErrors } = require('@jamietanna/spectral-test-harness')
+
+describe('GOV UK Rules', () => {
+  test('Complete OpenAPI 3.0.0 document passes', async () => {
```
Comment (jamietanna) [inline govuk-public-api-rules/test/rules.test.js:4]: We'd spoken offline about Swagger 2.0 too - should we add it now?

File: govuk-public-api-rules/package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "govuk-public-api-rules",
```
Comment (jamietanna) [inline govuk-public-api-rules/package.json:2]: Should it be clear these are spectral rules?

File: govuk-public-api-rules/package.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
```
Comment (jamietanna) [inline govuk-public-api-rules/package.json:1]: Should we add tags?

**Comment by nahteb** [comment]
Didn't we have some other tests to catch invalid version numbers and things? Or is that going to go in a separate PR?

File: govuk-public-api-rules/ruleset.yaml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+extends:
```
Comment (nahteb) [inline govuk-public-api-rules/ruleset.yaml:1]: I think in the other repo we decided to prefer this layout to avoid introducing extra whitespace changes in the future

```suggestion
extends:
  - "spectral:oas"
  - all
```

Comment (jamietanna) [reply govuk-public-api-rules/ruleset.yaml:1]: Good thinking - this is actually an odd case, where we've got a list inside a list, so I made the choice that this was a better version than:

```yaml
extends:
  -  - spectral:oas
     - all
```

File: govuk-public-api-rules/package.json (hunk 2 above)
Comment (jamietanna) [reply govuk-public-api-rules/package.json:2]: Perhaps:

```suggestion
  "name": "spectral-ruleset-govuk-public-api",
```

(We'd also want to rename the directory name / any references)

**Comment by jamietanna** [comment]
> Didn't we have some other tests to catch invalid version numbers and things? Or is that going to go in a separate PR?

Ah that was one we were just writing to trial writing new rules, rather than it being a "real" rule. I'd be happy to get us to reimplement it, but would say probably as a separate PR, so this one is just getting us ready to go to publish rules

File: spectral-ruleset-govuk-public/ruleset.yaml [hunk 5]
```diff
@@ -0,0 +1,2 @@
+extends:
+  - [spectral:oas, all]
```
Comment (jamietanna) [inline spectral-ruleset-govuk-public/ruleset.yaml:2]: Did we want this?

```suggestion
  - - spectral:oas
    - all
```

(Unlike the usual request of reducing diff noise, this array is only going to contain two items)

Comment (nahteb) [reply spectral-ruleset-govuk-public/ruleset.yaml:2]: I think stick with your original decision to leave as is if it's not going to change?

Comment (jamietanna) [reply spectral-ruleset-govuk-public/ruleset.yaml:2]: Sounds like a plan :+1:

File: spectral-ruleset-govuk-public/test/testdata/complete/valid-2.0.0.yaml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+swagger: "2.0"
+info:
+  title: ''
+  version: ''
```
Comment (jamietanna) [inline spectral-ruleset-govuk-public/test/testdata/complete/valid-2.0.0.yaml:4]: inconsistent quotes - did we want to align on one or the other?

**Review by nahteb — APPROVED** [review]
